### PR TITLE
#160577443 Implement articles CRUD

### PR DESCRIPTION
**Pull request by codingedward**
## What does this PR do?
- It adds support for creating, reading, updating and deleting articles.
- Adds a home page. 

## Description of Task to be completed?
- Add an editor supporting creating and editing of an article

## How should this be manually tested?
- Clone this repository by running `git clone "https://github.com/andela/ah-shakas-frontend.git"`
- Change working directory to project folder `ah-shakas-frontend`
- Install dependencies through `yarn install`
- Create a `.env` file with the following values
  ```
  REACT_APP_BASE_URL=
  REACT_APP_UPLOAD_BASE_URL=
  REACT_APP_CLOUD_NAME=
  REACT_APP_UPLOAD_PRESET=
  ```
  The `CLOUD_NAME` and the `UPLOAD_PRESET` are gotten from your cloudinary account.

## Any background context you want to provide?
- The create/edit of an article is achieved using a WYSIWYG editor called [Dante2](https://github.com/michelson/dante2)

## What are the relevant pivotal tracker stories?
[#160577443](https://www.pivotaltracker.com/story/show/160577443)

## Packages installed
- react-animated-slider
- Dante2@next
- prismjs

## Deployment note
N/A

## Related PRs branch | PR (branch_name) | (pr_link)
Branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()

## Todos
- [x] Raise PR
- [ ] Test
- [ ] Documentation

[#](link)

## Screenshots (if appropriate)
The homepage: 
screen shot 2018-11-13 at 11 33 08
screen shot 2018-11-13 at 11 33 19

The edit/create article page:
screen shot 2018-11-13 at 11 45 38

### CODE REVIEW

File: src/__tests__/actions/articles.action.test.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    store.clearActions();
+  });
+
+  it('send post article action', () => {
+    store.dispatch(postArticle(article));
+    expect(store.getActions()[0].type).toBe('POST_ARTICLE');
+  });
+});
+
+describe('Test user fetch single article action', () => {
```
Comment (Arthur236) [inline src/__tests__/actions/articles.action.test.js]: You dont need to have so many describe blocks in your tests. One is enough. Just make sure that the `it` blocks accurately indicate what the test does. Something like:

```
describe('Test articles actions', () => {
  it('dispatches POST_ARTICLE action', () => {
  });

  it('dispatches FETCH_ARTICLE action', () => {
  });
});
```

File: src/__tests__/containers/HomePageComponent.test.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+};
+const wrapper = mount(<HomePageComponent {...props} store={store} />);
+
+describe('renders <HomePageComponent/>', () => {
+  it('tests if component did mount', () => {
+    const spy = jest.spyOn(HomePageComponent.prototype, 'componentDidMount');
+    wrapper.instance().componentDidMount();
+    expect(spy).toHaveBeenCalled();
+  });
+  it('tests render <Slider/>', () => {
```
Comment (Arthur236) [inline src/__tests__/containers/HomePageComponent.test.js]: Have a blank line to separate each `it` block

File: src/assets/styles/EditorComponent.scss [hunk 3]
```diff
@@ -0,0 +1,14 @@
+.publish-btn {
+  float: right;
+  cursor: pointer;
+}
+
+
```
Comment (Arthur236) [inline src/assets/styles/EditorComponent.scss]: Remove the extra line

File: src/index.js [hunk 4]
```diff
@@ -9,6 +9,7 @@ import * as serviceWorker from './serviceWorker';
 import storeConfig from './store/index';
 
 const history = createHistory();
+// const store = storeConfig();
```
Comment (Arthur236) [inline src/index.js]: This comment is not necessary

File: src/utils/uploader.util.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import config from '../config';
+
+const { CLOUD_NAME, UPLOAD_PRESET } = config;
+
+export default (upload) => {
```
Comment (Arthur236) [inline src/utils/uploader.util.js:5]: Why not use axios for this?

**Comment by Arthur236** [comment]
Delete this file `src/actions/index.js`

File: src/utils/uploader.util.js (hunk 5 above)
Comment (codingedward) [reply src/utils/uploader.util.js:5]: Hi @Arthur236,

About this, the editor we are using uses the event we are dispatching in the `xhr.onreadystate` to update the progress of the upload (line 11).

File: src/index.js (hunk 4 above)
Comment (codingedward) [reply src/index.js]: Noted.

File: src/assets/styles/EditorComponent.scss (hunk 3 above)
Comment (codingedward) [reply src/assets/styles/EditorComponent.scss]: Noted.

File: src/__tests__/containers/HomePageComponent.test.js (hunk 2 above)
Comment (codingedward) [reply src/__tests__/containers/HomePageComponent.test.js]: Noted 👍

File: src/__tests__/actions/articles.action.test.js (hunk 1 above)
Comment (codingedward) [reply src/__tests__/actions/articles.action.test.js]: Noted 👍

File: src/utils/uploader.util.js (hunk 5 above)
Comment (Arthur236) [reply src/utils/uploader.util.js:5]: Alright 👍

File: src/assets/styles/HomePageComponent.scss [hunk 6]
```diff
@@ -0,0 +1,419 @@
...
+	opacity: 0;
+}
+
+.slide section span strong {
+	-webkit-transform: translateY(10px);
+					transform: translateY(10px);
+	opacity: 0;
+}
+
+.slide.animateIn.previous h1,
```
[the remark this answers is not in the document]
Comment (codingedward) [reply src/assets/styles/HomePageComponent.scss]: This is used by the animation library `react-animated-slider`

File: src/App.jsx [hunk 7]
```diff
@@ -1,24 +1,28 @@
...
   <div>
     <Switch>
-      <Route path="/" exact component={HomePage} />
+      <Route path="/" exact component={HomePageContainer} />
       <Route path="/login" exact component={LoginContainer} />
       <Route path="/signup" exact component={SignUpContainer} />
       <Route path="/forgot_password" exact component={ForgotPasswordContainer} />
       <Route path="/reset_password/:token" exact component={ResetPasswordContainer} />
-      <Route path="/dashboard" exact component={DashboardComponent} />
+      <Route path="/reset_password/:token" exact component={withAlert(ResetPasswordContainer)} />
```
Comment (Arthur236) [inline src/App.jsx:22]: This route is similar to the one above.
Remove one of these

File: src/__tests__/App.test.js [hunk 8]
```diff
@@ -1,9 +1,8 @@
 import React from 'react';
```
Comment (Arthur236) [inline src/__tests__/App.test.js]: We do not need this test file as we already have one in the components directory

**Comment by Arthur236** [comment]
You can delete this file since it serves no purpose `src/actions/index.js`

**Review by Arthur236 — APPROVED** [review]
Looks good 👍

**Pull request merged**